### DISCUSSION TITLE
Can't find cluster

### DISCUSSION THREAD

**Discussion by Ping-lin14** [stuart-lab/signac#651] [Q&A] [answered]
Hi
I followed the Joint **_RNA and ATAC analysis: 10x multiomic_** tutorial to perform clustering, where I skipped annotating the cells, and the rest are based on the tutorial.
However, after I visualized UMAP, I got the picture below.
![image](https://user-images.githubusercontent.com/69965537/120883603-a2522180-c610-11eb-897c-590b9a923916.png)

I tried to execute the `FindClusters` function after `RunUMAP` function, but received an error message.
```
Error in FindClusters.Seurat(object = tc, verbose = FALSE, algorithm = 3,  : 
  Provided graph.name does not correspond to a graph object.
```
Is there a solution.

My code:.
```
tc <- FindMultiModalNeighbors(
  object = tc,
  reduction.list = list("pca", "lsi"), 
  dims.list = list(1:50, 2:40),
  modality.weight.name = c("RNA.weight", "ATAC.weght"),
  verbose = TRUE
)

# build a joint UMAP visualization
tc <- RunUMAP(
  object = tc,
  nn.name = "weighted.nn",
  assay = "ATAC",
  verbose = TRUE
)

tc <- FindClusters(object = tc, verbose = FALSE, algorithm = 3)

DimPlot(tc, label = TRUE, repel = TRUE, reduction = "umap") + NoLegend()
```

**Comment by timoast** [comment, marked as answer]
You need to run `FindNeighbors()` before running `FindClusters()`
